vortex-file/src/v2/file.rs

```diff
@@ -41,94 +43,46 @@ impl<R: VortexReadAt + Unpin> VortexFile<R> {
     }
 
     /// Performs a scan operation over the file.
-    pub fn scan(self, scan: Arc<Scan>) -> VortexResult<impl ArrayStream + 'static> {
-        // Create a shared reader for the scan.
-        let reader: Arc<dyn LayoutReader> = self
-            .layout
-            .reader(self.segments.reader(), self.ctx.clone())?;
+    pub fn scan(&self, scan: Arc<Scan>) -> VortexResult<impl ArrayStream + 'static + use<'_, I>> {
```

Comment:
I'm confused by the lifetime signature here. How does `use<...>` change anything if the lifetime has to be 'static?